From ff9216a250488e5d08410f65e6cf13288fb83bf6 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 25 May 2021 10:14:16 +0300
Subject: [PATCH 08/10] clocksource: stm: Correct IRQ affinity

commit 8c0e7a3244ce870d3a41dc148f75b67ed99f1990 from
https://source.codeaurora.org/external/autobsps32/linux

Replace IRQF_IRQPOLL with IRQF_PERCPU. This will avoid the
schedule of STM interrupts on cores that are not assigned
to this module.

Upstream-Status: pending

Issue: ALB-6311
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/clocksource/timer-fsl-stm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clocksource/timer-fsl-stm.c b/drivers/clocksource/timer-fsl-stm.c
index 5f8db335923c..cc383ef538b2 100644
--- a/drivers/clocksource/timer-fsl-stm.c
+++ b/drivers/clocksource/timer-fsl-stm.c
@@ -256,7 +256,7 @@ static int stm_clockevent_init(struct stm_timer *stm, unsigned long rate,
 	stm->clockevent_stm.irq = irq;
 
 	stm->stm_timer_irq.name = STM_TIMER_NAME;
-	stm->stm_timer_irq.flags = IRQF_TIMER | IRQF_IRQPOLL;
+	stm->stm_timer_irq.flags = IRQF_TIMER | IRQF_PERCPU;
 	stm->stm_timer_irq.handler = stm_timer_interrupt;
 	stm->stm_timer_irq.dev_id = &stm->clockevent_stm;
 
-- 
2.25.1

